### PR TITLE
Bluetooth request info

### LINKED ISSUE #239: Bluetooth pop-up request missing information
**Describe the bug**
Currently if the user has BT off after initial installation the application has an alert dialog "CoEpi wants to turn on Bluetooth" with options to Deny or Allow. The need for this is not clear or included in the popup, which goes against the "Tenets of working with Android permissions" in that it is not clear why the user needs to give access to BT. Selecting Deny clears the view without any further explanation of how the app will be diminished. 
Even though the BT permissions are generally tied to the GPS location permissions it is still necessary to clarify why BT needs to be enabled in this alert dialog.

**To Reproduce**
1. Install application
2. Accept Location permissions
3. Turn off Bluetooth on phone
4. See alert dialog

**Expected behavior**
An alert dialog is displayed still at step 4, but has additional information.
"CoEpi wants to turn on Bluetooth:
_Brief explanation of how BT is used in this app and how the app will be different if the user denies access._
Deny Allow"

**Screenshots**
Bluetooth-Request

**Additional context**
Phone: Pixel 3
OS: Android 10
App Version: https://github.com/Co-Epi/app-android/tree/559e0a273b5eade24436d12e9044b27f6bcb4222

### PR DESCRIPTION
**Pull request by Joshua-McCord**
App requests permission per the Google Documentation. Explains the use after the first denial of features. If the user presses Deny and Don't Allow Again, it will no longer prompt. (Still in line with Google Documentation). 

device-2020-06-17-155956

### CODE REVIEW

**Comment by ivnsch** [comment]
https://developer.android.com/training/permissions/requesting#explain ?

**Comment by mellowgeek** [comment]
With the update to include the rationale dialog, I think we should change the buttons to be more clear that you're going to prompt for permissions in that dialog. The two examples I've seen are either give a single option that says "OK", or stick with the two options and instead of "Allow/Don't Allow", use something like "Retry/I'm sure"

**Review by ivnsch — CHANGES_REQUESTED** [review]
(Slack)

File: app/src/main/java/org/coepi/android/ui/onboarding/OnboardingPermissionsChecker.kt [hunk 1]
```diff
@@ -28,10 +32,22 @@ class OnboardingPermissionsChecker {
         val hasAllPermissions = permissions.all {
             checkSelfPermission(activity, it) == PERMISSION_GRANTED
         }
-        if (hasAllPermissions) {
-            observable.onNext(true)
-        } else {
-            requestPermissions(activity, permissions, requestCode)
+        when {
+            hasAllPermissions -> observable.onNext(true)
+            shouldShowRequestPermissionRationale(activity, ACCESS_COARSE_LOCATION) ->
```
Comment (ivnsch) [inline app/src/main/java/org/coepi/android/ui/onboarding/OnboardingPermissionsChecker.kt]: Why only for `ACCESS_COARSE_LOCATION`?

**Comment by ivnsch** [comment]
As per https://github.com/Co-Epi/app-android/issues/239#issuecomment-647987762, would it make sense to request the permissions after the onboarding?

**Pull request merged**